sky/clouds/aws.py

```diff
@@ -399,16 +401,31 @@ def make_deploy_resources_variables(self,
 
         user_security_group = skypilot_config.get_nested(
             ('aws', 'security_group_name'), None)
-        if resources.ports is not None:
-            # Already checked in Resources._try_validate_ports
-            assert user_security_group is None
-            security_group = USER_PORTS_SECURITY_GROUP_NAME.format(
-                cluster_name_on_cloud)
-        elif user_security_group is not None:
-            assert resources.ports is None
-            security_group = user_security_group
-        else:
+        if user_security_group is not None and not isinstance(
+                user_security_group, str):
```

Comment:
nit: this seems more direct to me
```suggestion
        if user_security_group is not None and isinstance(user_security_group, list):
```